internal/freelist/shared.go

```diff
@@ -0,0 +1,280 @@
+package freelist
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"unsafe"
+
+	"go.etcd.io/bbolt/internal/common"
+)
+
+type sharedInterface interface {
+	// Init initializes this freelist with the given list of pages.
+	Init(ids common.Pgids)
+	// Count returns the number of free and pending pages.
+	Count() int
+	// FreeCount returns the number of free pages.
+	FreeCount() int
+	// FreePageIds returns all free pages.
+	FreePageIds() common.Pgids
+	// mergeSpans is merging the given pages into the freelist
+	mergeSpans(ids common.Pgids)
+}
```

Comment:
You can add `mergeSpans` into `Interface`
```suggestion
```

Reply:
done